### PR TITLE
Missing Support For Custom Hostname Update Properties

### LINKED ISSUE #179: Missing Support For Custom Hostname Update Properties
When creating a custom hostname using the addHostname function at https://github.com/cloudflare/cloudflare-php/blob/master/src/Endpoints/CustomHostnames.php#L38, there is no support for the `wildcard`, `bundle_method`, `custom_certificate` or `custom_key fields` as documented at https://api.cloudflare.com/#custom-hostname-for-a-zone-edit-custom-hostname.

### PR DESCRIPTION
**Pull request by phily245**
As mentioned in #179, there are some missing properties from custom hostname creation (shown at https://api.cloudflare.com/#custom-hostname-for-a-zone-edit-custom-hostname). This PR attempts to add these in.

Closes #179

### CODE REVIEW

**Comment by jacobbednarz** [comment]
similar to your other PR, the lint failures will need addressing here.

File: src/Endpoints/CustomHostnames.php [hunk 1]
```diff
@@ -120,16 +120,33 @@ public function getHostname(string $zoneID, string $hostnameID)
 
     /**
      * @SuppressWarnings(PHPMD.BooleanArgumentFlag)
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
```
Comment (phily245) [inline src/Endpoints/CustomHostnames.php:124]: @jacobbednarz I'm unsure how is the best way to refactor this in the way the SDK has been built. Would a protected method to build the query be acceptable, or should I let the PHPMD suppression stay here or any other suggestions you may have?

File: tests/Endpoints/CustomHostnamesTest.php [hunk 2]
```diff
@@ -148,7 +155,20 @@ public function testUpdateHostname()
             'http3' => 'on',
             'min_tls_version' => '1.2'
         ];
-        $result = $zones->updateHostname('023e105f4ecef8ad9ca31a8372d0c353', '0d89c70d-ad9f-4843-b99f-6cc0252067e9', 'http', 'dv', $sslSettings, 'origin.example.com');
+        $result = $zones->updateHostname(
```
Comment (phily245) [inline tests/Endpoints/CustomHostnamesTest.php:158]:
```suggestion

        $result = $zones->updateHostname(
```

File: tests/Endpoints/CustomHostnamesTest.php [hunk 3]
```diff
@@ -148,7 +155,21 @@ public function testUpdateHostname()
             'http3' => 'on',
             'min_tls_version' => '1.2'
         ];
-        $result = $zones->updateHostname('023e105f4ecef8ad9ca31a8372d0c353', '0d89c70d-ad9f-4843-b99f-6cc0252067e9', 'http', 'dv', $sslSettings, 'origin.example.com');
+
+        $result = $zones->updateHostname(
```
Comment (phily245) [inline tests/Endpoints/CustomHostnamesTest.php:159]:
```suggestion
        $result = $zones->updateHostname(
```

File: tests/Endpoints/CustomHostnamesTest.php (hunk 2 above)
Comment (phily245) [reply tests/Endpoints/CustomHostnamesTest.php:158]:
```suggestion

        $result = $zones->updateHostname(
```

**Comment by phily245** [comment]
@jacobbednarz the failing lint has been sorted here too, it took a bit longer because the pipeline was randomly failing to pull from dockerhub for some reason.

**Comment by phily245** [comment]
> @jacobbednarz the failing lint has been sorted here too, it took a bit longer because the pipeline was randomly failing to pull from dockerhub for some reason.

E.g. https://github.com/cloudflare/cloudflare-php/runs/2884256252?check_suite_focus=true

File: src/Endpoints/CustomHostnames.php (hunk 1 above)
Comment (jacobbednarz) [reply src/Endpoints/CustomHostnames.php:147]: I'm fine with the suppression here as it's pretty clear what the intention is.

**Comment by jacobbednarz** [comment]
Thanking you! 👏

**Pull request merged**